src/api/client/resolvers.ts

```diff
@@ -152,10 +152,99 @@ export const queryNodeStitchingResolvers = (
       RemoveQueryNodeChannelViewsField,
     ]),
     // mixed queries
-    search: createResolverWithTransforms(queryNodeSchema, 'search', [
-      RemoveQueryNodeViewsField,
-      RemoveQueryNodeFollowsField,
-    ]),
+    search: async (parent, args, context, info) => {
+      const searchResolver = createResolverWithTransforms(queryNodeSchema, 'search', [
+        RemoveQueryNodeViewsField,
+        RemoveQueryNodeFollowsField,
+        RemoveQueryNodeChannelViewsField,
+      ])
+      const search = await searchResolver(parent, args, context, info)
+
+      const channelIdList = search
+        .filter((result: SearchFtsOutput) => result.item.__typename === 'Channel')
+        .map((result: SearchFtsOutput) => result.item.id)
+
+      const batchedChannelFollowsResolver = createResolverWithTransforms(
+        orionSchema,
+        ORION_BATCHED_FOLLOWS_QUERY_NAME,
+        [TransformBatchedOrionFollowsField],
+        // operationName has to be manually kept in sync with the query name used
+        BATCHED_FOLLOWS_VIEWS_QUERY_NAME
+      )
+      const batchedChannelFollows = await batchedChannelFollowsResolver(
+        parent,
+        {
+          channelIdList,
+        },
+        context,
+        info
+      )
+
+      const batchedChannelViewsResolver = createResolverWithTransforms(
+        orionSchema,
+        ORION_BATCHED_CHANNEL_VIEWS_QUERY_NAME,
+        [TransformBatchedChannelOrionViewsField],
+        // operationName has to be manually kept in sync with the query name used
+        BATCHED_CHANNEL_VIEWS_QUERY_NAME
+      )
+      const batchedChannelViews = await batchedChannelViewsResolver(
```

Comment:
We should actually await both of those at the same time to reduce the request time. Like so:
```ts
const batchedChannelFollowsPromise = batchedChannelFollowsResolver(...)
const batchedChannelViewsPromise = batchedChannelViewsResolver(...)
const [batchedChannelFollows, batchedChannelViews] = await Promise.all([
  batchedChannelFollowsPromise,
  batchedChannelViewsPromise
])
```

I think the current resolver for `channels` query has the same problem, it'd be great if you could fix that as well

Reply:
good idea, I'll refactor this